metrotranscom/doorway
# release: listing duplication endpoint

This PR addresses [#(799)](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/metrotranscom/doorway/799)


## Description

Releases:
https://github.com/bloom-housing/bloom/pull/4313
https://github.com/bloom-housing/bloom/pull/4307
https://github.com/bloom-housing/bloom/pull/4314

## How Can This Be Tested/Reviewed?

Run partner portal and api and login as a admin user.
Have the listing id for the listing you would like to duplicate.
Fill out the payload for listings/duplicate. Set storedListing.id to the above.
Cases:

Set name to a unique name, set includeUnits to true. Should create a new draft listing with the name given which is otherwise identical to the original listing, including the units.
Set name to a unique name, set includeUnits to false. Should create a new draft listing with the name given which is otherwise identical to the original listing, excluding the units.
Set name to a the same name as the original listing. Should error with 400 BadRequest and a message 'New listing name must be unique'.
You can view the newly created in the Partners Portal

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
